lib/puppet/provider/apt_key/apt_key.rb

```diff
@@ -136,6 +136,18 @@ def tempfile(content)
     file = Tempfile.new('apt_key')
     file.write content
     file.close
+    #confirm that the fingerprint from the file, matches the long key that is in the manifest
+    if name.size == 40
+      if File.executable? "/usr/bin/gpg"
+        extracted_key = execute(["/usr/bin/gpg --with-fingerprint --with-colons #{file.path} | awk -F: '/^fpr:/ { print $10 }'"], :failonfail => false) 
```

Comment:
I think we should define this as a `commands`, just like we're doing with `apt-get`. It gets you some nice and saner behaviour too.